src/neo4j/_auth_management.py

```diff
@@ -52,8 +59,37 @@ class ExpiringAuth:
 
     .. versionadded:: 5.8
     """
-    auth: _TAuth
-    expires_in: t.Optional[float] = None
+    auth: "_TAuth"
+    expires_at: t.Optional[float] = None
+
+    def expires_in(self, seconds: float) -> "ExpiringAuth":
```

Comment:
That's to defer the evaluation of the type hint as the class is not yet defined (we're in it's definition).

This could also be achieved with `from __future__ import annotations` (or by using Python 3.10+ where that's the default behavior), but we have to keep support for Python 3.7 and using the `__future__` import breaks the docs generation tool (see [line 19 of the same file](https://github.com/neo4j/neo4j-python-driver/pull/928/files#diff-afe23d99150a31c4a8b55a545528304f4e27b3915d6301b75145e2874677fbb7R19)).

Reply:
Many thanks.